### PR TITLE
ref(onboarding): Remove multi-select platform for new orgs - [TET-664]

### PR DESCRIPTION
**Pull request by priscilawebdev**
Removing the multi-select platform from the onboarding of new organizations. The current pattern is sub-optimal and generates a lot of empty projects that people then don’t really know how to get back to. 

**Required Feature Flag:**

`organizations:onboarding-remove-multiselect-platform`

**Preview:**

https://user-images.githubusercontent.com/29228205/215111319-8b402e13-2691-41fb-a23e-afd1471d1acd.mov

**PS:**

As soon as we have everything approved and merged, tests will follow.

### CODE REVIEW

File: static/app/routes.tsx [hunk 1]
```diff
@@ -242,7 +242,7 @@ function buildRoutes() {
           <IndexRedirect to="welcome/" />
           <Route
             path=":step/"
-            component={make(() => import('sentry/views/onboarding/onboarding'))}
```
Comment (priscilawebdev) [inline static/app/routes.tsx:245]: minor improvement. Please let me know if this change if fine when reviewing this PR

File: static/app/views/onboarding/components/createProjectsFooter.tsx [hunk 2]
```diff
@@ -144,6 +168,7 @@ const ButtonWrapper = styled(motion.div)`
   height: 100%;
   align-items: center;
   margin-right: ${space(4)};
+  margin-left: ${space(4)};
```
Comment (priscilawebdev) [inline static/app/views/onboarding/components/createProjectsFooter.tsx:171]: this change is not behind feature flag, but styles look better

File: static/app/views/onboarding/components/createProjectsFooter.tsx [hunk 3]
```diff
@@ -156,10 +181,22 @@ const SelectedPlatformIcon = styled(PlatformIcon)`
...
 const PlatformSelected = styled('div')`
   margin-top: ${space(1)};
+  display: grid;
+  grid-template-columns: 1fr max-content max-content;
+  align-items: center;
 `;
 
 const ClearButton = styled(Button)`
   margin-left: ${space(2)};
+  padding: 0;
```
Comment (priscilawebdev) [inline static/app/views/onboarding/components/createProjectsFooter.tsx:197]: this change is not behind a feature flag, but styles look better

File: static/app/views/onboarding/components/heartbeatFooter/useHeartbeat.tsx [hunk 4]
```diff
@@ -72,26 +67,16 @@ export function useHeartbeat(
...
+    onSuccess: data => {
+      setFirstIssue(data.find((issue: Group) => issue.firstSeen === firstError));
+    },
+  });
 
   const firstErrorReceived = firstIssue ?? !!firstError;
-  const isFetchedAfterMount = eventIsFetchedAfterMount && sessionIsFetchedAfterMount;
-  const loading = eventIsLoading || sessionIsLoading || !isFetchedAfterMount;
-
-  useEffect(() => {
```
Comment (priscilawebdev) [inline static/app/views/onboarding/components/heartbeatFooter/useHeartbeat.tsx:90]: was looking forward to remove this recent `hack`

File: static/app/views/onboarding/components/projectSidebarSection.tsx [hunk 5]
```diff
@@ -14,75 +12,21 @@ import space from 'sentry/styles/space';
...
   activeProject: Project | null;
   checkProjectHasFirstEvent: (project: Project) => boolean;
-  hasHeartbeatFooter: boolean;
   projects: Project[];
   selectProject: (newProjectId: string) => void;
   // A map from selected platform keys to the projects created by onboarding.
   selectedPlatformToProjectIdMap: {[key in PlatformKey]?: string};
 };
 
-function NewProjectSideBarSection({
```
Comment (priscilawebdev) [inline static/app/views/onboarding/components/projectSidebarSection.tsx:29]: Hearbeat is no longer needed here

File: static/app/views/onboarding/index.tsx [hunk 6]
```diff
@@ -226,14 +236,14 @@ function Onboarding(props: Props) {
   );
 }
 
-const Container = styled('div')<{hasFooter: boolean; hasHeartbeatFooter: boolean}>`
+const Container = styled('div')<{hasFooter: boolean; heartbeatFooter: boolean}>`
```
Comment (priscilawebdev) [inline static/app/views/onboarding/index.tsx:239]: just updated the name

File: static/app/views/onboarding/setupDocs.tsx [hunk 7]
```diff
@@ -45,7 +45,7 @@ type Props = {
   loadingProjects?: boolean;
 } & StepProps;
 
-function ProjecDocs(props: {
+function ProjectDocs(props: {
```
Comment (priscilawebdev) [inline static/app/views/onboarding/setupDocs.tsx:48]: there was a typo here. Please let me know if this change is fine when reviewing

**Review by Jesse-Box — APPROVED** [review]
From a purely design perspective, I think this is good to go :)

**Review by matejminar — APPROVED** [review]

**Review by scefali — COMMENTED** [review]
@priscilawebdev I'm surprised this PR doesn't remove or touch the `MultiPlatformPicker` component since that's what we use to select multiple platforms

**Comment by priscilawebdev** [comment]
@scefali we not removing the `MultiPlatformPicker` just yet. These new changes are behind a feature flag and we would like to test them internally before rolling them out to customers. 

The next step would be the removal of the `MultiplatformPicker` component 😉

**Pull request merged**